expectation.go

```
@@ -193,9 +193,23 @@ func (e *expectation[T]) defaultFailureReport(reporter any, matcher any, opts ..
 		gf = fmt.Sprintf("%v", e.subject)
 	}
 
+	// Check if exp is nil using reflection to handle interface types properly
+	isNil := func(x any) bool {
+		if x == nil {
+			return true
+		}
+		v := reflect.ValueOf(x)
+		switch v.Kind() {
```

golangci-lint (2.11.4)

[error] missing cases in switch of type reflect.Kind: reflect.Invalid, reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr, reflect.Float32, reflect.Float64, reflect.Complex64, reflect.Complex128, reflect.Array, reflect.String, reflect.Struct, reflect.UnsafePointer

(exhaustive)